home/management/commands/updatecourses.py

```diff
@@ -0,0 +1,99 @@
+                            name=umdio_course['course_id'],
+                            department=umdio_course['dept_id'],
+                            course_number=umdio_course['course_id'][4:],
+                            title=umdio_course['name'],
+                            credits=umdio_course['credits'],
+                            description=umdio_course["description"]
+                        )
+
+                        course.save()
+                        self.total_num_new_courses += 1
+
+                    self._professors(course, semester)
+                    print(course)
+
+                kwargs["page"] += 1
+                course_data = requests.get("https://api.umd.io/v1/courses", params=kwargs).json()
+
+        print(f"\n** New Courses Created: {self.total_num_new_courses} **")
+        print(f"** New Professors Created: {self.total_num_new_professors} **")
+
+        runtime = datetime.now() - t_start
+        print(f"Runtime: {round(runtime.seconds / 60, 2)} minutes")
+
+    def _professors(self, course: Course, semester: Semester):
+        kwargs = {"course_id": course.name}
+        umdio_professors = requests.get("https://api.umd.io/v1/professors", params=kwargs).json()
+
+        # if no professors were found for `course` during `semester`
+        if isinstance(umdio_professors, dict) and 'error_code' in umdio_professors.keys():
+            return
+
+        for umdio_professor in umdio_professors:
+            if "instructor" in umdio_professor['name'].lower():
+                continue
+
+            professors = Professor.verified.all()
+
+            if not professors.filter(name=umdio_professor['name']).first():
+                # To make our lives easier, attempt to automatically verify the professor
+                # following a similar process to that in admin.py
+                professor = Professor(name=umdio_professor['name'], type=Professor.Type.PROFESSOR)
+                similar_professors = Professor.find_similar(professor.name, 70)
+                split_name = professor.name.strip().split()
+                new_slug = split_name[-1].lower()
+                valid_slug = True
+
+                if professors.filter(slug=new_slug).exists():
+                    new_slug = f"{split_name[-1]}_{split_name[0]}".lower()
+                    if professors.filter(slug=new_slug).exists():
+                        valid_slug = False
+
+                if len(similar_professors) == 0 and valid_slug:
+                    professor.slug = new_slug
+                    professor.status = Professor.Status.VERIFIED
+
+                professor.save()
+                self.total_num_new_professors += 1
+
+            for entry in umdio_professor['taught']:
+                if entry['course_id'] == course.name and Semester(entry['semester']) == semester:
+                    ProfessorCourse.objects.create(course=course, professor=professor, recent_semester=semester)
```

Comment:
if `not professors.filter(name=umdio_professor['name']).first()` is false above, then `professor` is never set and this errors. Run `python3 manage.py updatecourses 202201` to reproduce.

Reply:
The latest commit addresses this. I also realized that I was using `.first()` in a dangerous way that would've defaulted to the first professor in a query with more than one result. This has also been fixed.